api/components/button.schema.json

```diff
@@ -30,10 +30,10 @@
             "$ref": "defs/listener.schema.json"
         },
         "leftIcon": {
-            "$ref": "defs/icon.schema.json"
+            "$ref": "icon.schema.json"
```

Comment:
Since the icon is now a component (not a property), this shouldn't be a ref but more a `"type": "component", "enum":...  ` like the subcomponent in the "text" component.